dspace-server-webapp/src/test/java/org/dspace/app/rest/ResearcherProfileRestRepositoryIT.java

```diff
@@ -0,0 +1,1093 @@
+            .andExpect(status().isCreated());
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+        getClient(adminToken).perform(delete("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNoContent());
+
+        getClient(adminToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNotFound());
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNotFound());
+    }
+
+    /**
+     * Verify that an user can delete his profile using the delete endpoint even if
+     * was created by an admin.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDeleteProfileCreatedByAnAdmin() throws Exception {
+
+        String id = user.getID().toString();
+
+        String adminToken = getAuthToken(admin.getEmail(), password);
+        String userToken = getAuthToken(user.getEmail(), password);
+
+        getClient(adminToken).perform(post("/api/eperson/profiles/")
+            .param("eperson", id)
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isCreated());
+
+        getClient(adminToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+        getClient(userToken).perform(delete("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNoContent());
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNotFound());
+
+        getClient(adminToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isNotFound());
+
+    }
+
+    /**
+     * Verify that a standard user can't call the delete endpoint to delete a
+     * researcher profile related to another user.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDeleteWithoutOwnUser() throws Exception {
+
+        String id = user.getID().toString();
+
+        String userToken = getAuthToken(user.getEmail(), password);
+        String anotherUserToken = getAuthToken(anotherUser.getEmail(), password);
+
+        getClient(userToken).perform(post("/api/eperson/profiles/")
+            .contentType(MediaType.APPLICATION_JSON_VALUE))
+            .andExpect(status().isCreated());
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+        getClient(anotherUserToken).perform(delete("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isForbidden());
+
+        getClient(userToken).perform(get("/api/eperson/profiles/{id}", id))
+            .andExpect(status().isOk());
+
+    }
+
+    /**
+     * Verify that an user can change the profile visibility using the patch endpoint.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testPatchToChangeVisibleAttribute() throws Exception {
```

Comment:
I might be missing it, but I don't see a test that verifies the behavior of `visible`.  I.e. a test proving that setting it to `true` means an anonymous user can access while setting to `false` means anonymous cannot access (but the user themselves still can & so can admins)